qmanager/policies/base/queue_policy_base.hpp

```diff
@@ -379,29 +455,44 @@ class queue_policy_base_t : public detail::queue_policy_base_impl_t,
      *  inserted into the pending job queue or a job was removed from
      *  the running job queue so that its resource was released.
      */
-    bool is_schedulable ();
+    bool is_schedulable ()
+    {
+        return m_schedulable;
+    }
 
     /*! Set this queue's schedulability. After this call,
      *  is_schedulable() will return the newly set schedulability
      *  until a new job or resource event occurs.
      */
-    void set_schedulability (bool schedulable);
+    void set_schedulability (bool schedulable)
+    {
+        m_schedulable = schedulable;
+    }
 
     /*! Return true if the job state of this queue has changed
      *  as the result of the invocation of schedule loop and
      *  and/or of other conditions.
      */
-    bool is_scheduled ();
+    bool is_scheduled ()
+    {
+        return m_scheduled;
+    }
 
     /*! Reset this queue's "scheduled" state.
      */
-    void reset_scheduled ();
+    void reset_scheduled ()
+    {
+        m_scheduled = false;
+    }
 
     /*! Implement queue_adapter_base_t's pure virtual method
      *  so that this queue can be adapted for use within high-level
      *  resource API. Return true if the scheduling loop is active.
      */
-    virtual bool is_sched_loop_active ();
+    bool is_sched_loop_active ()
```

Comment:
The ones that were virtual need to stay virtual. Looks like there's at least one more, but we need to be careful about that, could make the behavior very strange otherwise.

```suggestion
    virtual bool is_sched_loop_active ()
```

Reply:
This should be fixed. There was one other function.